connect/basic-auth-extension/src/main/java/org/apache/kafka/connect/rest/basic/auth/extension/JaasBasicAuthFilter.java

```diff
@@ -39,7 +39,7 @@
 public class JaasBasicAuthFilter implements ContainerRequestFilter {
     private static final String CONNECT_LOGIN_MODULE = "KafkaConnect";
     static final String AUTHORIZATION = "Authorization";
-    private static final String TASK_REQUEST_PATTERN = "\\/?connectors\\/([^\\/]*)\\/tasks\\/?"; //\/?connectors\/([^\/]*)\/tasks\/?
+    private static final String TASK_REQUEST_PATTERN = "\\/?connectors\\/([^\\/]+)\\/tasks\\/?";
```

Comment:
I don't think the backslashes in this regex string are necessary; can it be simplified to `"/?connectors/[^/]+/tasks/?"`?

Reply:
Actually, why don't we initialize the final as a `Pattern` instance so that we don't have to parse the pattern with every request?

Reply:
That's a good addition too. We should definitely clean up the regex if possible too, though.

Reply:
yup. changed as well! Thanks for the suggestion